strax/context.py

```diff
@@ -415,12 +416,56 @@ def _set_plugin_config(self, p, run_id, tolerant=True):
                     parent_name = opt.parent_option_name
 
                     mes = (f'Cannot find "{parent_name}" among the options of the parent.'
-                           f' Either you specified by accident {option_name} as child option' 
+                           f' Either you specified by accident {option_name} as child option'
                            f' or you specified the wrong parent_option_name. Have you specified '
                            'the correct parent option name?')
                     assert parent_name in p.config, mes
                     p.config[parent_name] = option_value
 
+    def _config_hash(self):
+        """
+        Dump the current config to a hash as a sanity check for building
+        the _fixed_plugin_cache. If any item changes in the config, so
+        does this hash.
+        """
+        return hashlib.sha1(str(self.config).encode('ascii')).hexdigest()
+        # return strax.deterministic_hash(self.config)
+
+    def _plugins_are_cached(self, targets:  ty.Tuple[str]) -> bool:
+        """Check if all the requested targets are in the _fixed_plugin_cache"""
+        if self.context_config['use_per_run_defaults'] or self._fixed_plugin_cache is None:
+            # There is no point in caching if plugins (lineage) can
+            # change per run or the cache is empty.
+            return False
+
+        config_hash = self._config_hash()
+        if config_hash in self._fixed_plugin_cache:
+            targets_in_cache = [t in self._fixed_plugin_cache[config_hash] for t in targets]
+            if all(targets_in_cache):
+                return True
+
+        return False
+
+    def _plugins_to_cache(self, plugins: dict) -> None:
+        if self.context_config['use_per_run_defaults']:
+            # There is no point in caching if plugins (lineage) can change per run
+            return
+        if self._fixed_plugin_cache is None or self._config_hash() not in self._fixed_plugin_cache:
+            self._fixed_plugin_cache = {self._config_hash(): dict()}
```

Comment:
I assume Yossi's comment is also the reason why you are always making a new dictionary when the hash cannot be found?

Reply:
Very keenly spotted Daniel, I should have put a comment here.

So my suspicion is that the cache is only once created, at most twice if you change some option or register some plugin. I did think the likelihood that one was eating away memory for keeping the cache was greater than the change that someone was flipping between options often (although neither is very likely).

Reply:
> So my suspicion is that the cache is only once created, at most twice if you change some option or register some plugin. 

Yes, this would be my guess, too. As I said I am fine like it is but if you can implement Yossi's suggestion easily it would be a nice to have,

Reply:
I fully agree